ng-dev/pr/common/validation/validations.ts

```diff
@@ -102,3 +106,49 @@ export function assertPendingState(pullRequest: RawPullRequest) {
       throw PullRequestFailure.isMerged();
   }
 }
+
+/**
+ * Assert the pull request has all necessary CLAs signed.
+ * @throws {PullRequestFailure} if the pull request is missing a necessary CLA signature.
+ */
+export function assertSignedCla(pullRequest: PullRequestFromGithub) {
+  const passing = getStatusesForPullRequest(pullRequest).statuses.some(({name, status}) => {
+    return name === 'cla/google' && status === PullRequestStatus.PASSING;
+  });
+
+  if (passing) {
+    return;
+  }
+
+  throw PullRequestFailure.claUnsigned();
+}
+
+/**
+ * Assert the pull request has been marked ready for merge by the author.
+ * @throws {PullRequestFailure} if the pull request is missing the merge ready label.
+ */
+export function assertMergeReady(pullRequest: PullRequestFromGithub, config: PullRequestConfig) {
+  if (pullRequest.labels.nodes.some(({name}) => matchesPattern(name, config.mergeReadyLabel))) {
+    return true;
+  }
+  throw PullRequestFailure.notMergeReady();
+}
+
+/**
+ * Assert the pull request has been marked ready for merge by the author.
+ * @throws {PullRequestFailure} if the pull request is missing the merge ready label.
+ */
+export function assertPassingCi(pullRequest: PullRequestFromGithub) {
+  if (getStatusesForPullRequest(pullRequest).combinedStatus === PullRequestStatus.FAILING) {
+    throw PullRequestFailure.pendingCiJobs();
+  }
+  if (getStatusesForPullRequest(pullRequest).combinedStatus === PullRequestStatus.FAILING) {
```

Comment:
We should store the status in a variable for more readability and to save us from another invocation.